addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScanRule.java

```diff
@@ -231,16 +289,20 @@ private void checkObservedErrors(
         } else {
             noticesRisk = Alert.RISK_INFO;
         }
-        buildNoticesAlert(
-                        noticesRisk,
-                        getHeaderField(msg, HTTP_HEADER_CSP).get(0),
-                        csp,
-                        cspNoticesString)
-                .raise();
+        if (!cspNoticesString.isEmpty()) {
+            buildNoticesAlert(noticesRisk, CspUtils.HEADER_CSP, csp, cspNoticesString).raise();
+        }
     }
 
     private void checkWildcardSources(
-            List<String> allowedWildcardSources, HttpMessage msg, String csp) {
+            List<String> allowedWildcardSources, HttpMessage msg, String csp, boolean isMeta) {
+        if (isMeta) {
+            // frame-ancestors isn't applicable in META
+            allowedWildcardSources.remove("frame-ancestors");
+        }
```

Comment:
Why not let the caller do this? It would avoid the need to add an extra check here which is only applicable to one case.